File: 020/cpp/main.cpp
class Solution {
public:
    bool isValid(string s) {
        std::vector<char> v;
        for (auto &c: s) {
            if (c == '(' || c == '[' || c == '{') {
                v.push_back(c);
            } else if (c == ')' || c == ']' || c == '}') {
                if (!v.empty()) {
                    char c0 = v.back();
                    if ((c0 == '(' && c == ')') || (c0 == '[' && c == ']') || (c0 == '{' && c == '}')) {
                        v.pop_back();
                        continue;
                    }
                }
                return false;
            }
        }
        return v.empty();
    }
};

File: 020/rust/main.rs
impl Solution {
    pub fn is_valid(s: String) -> bool {
        let mut v = Vec::new();
        let s = s.as_bytes();
        for c in s {
            let t = *c as char;
            match t {
                '(' | '[' | '{' => v.push(t),
                ')' | ']' | '}' => {
                    if let Some(t0) = v.pop() {
                       if (t0 == '(' && t == ')') || (t0 == '[' && t == ']') || (t0 == '{' && t == '}') {
                           continue;
                       } 
                    }  
                    return false;
                }
                _ => (),
            }
        }
        return v.is_empty();
    }
}
